### PR TITLE
Add FSProjectStateResources

### PR DESCRIPTION
**Pull request by phlax**
for caching resources on state object, and to make it easier to review and test

### CODE REVIEW

File: pootle/apps/pootle_fs/resources.py [hunk 1]
```diff
@@ -47,3 +53,96 @@ def unsynced(self):
     @property
     def trackable_stores(self):
         return self.stores.exclude(obsolete=True).filter(fs__isnull=True)
+
+
+class FSProjectStateResources(object):
```
Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: An explanatory docstring would be appreciated.

File: pootle/apps/pootle_fs/resources.py [hunk 2]
```diff
@@ -47,3 +53,96 @@ def unsynced(self):
...
+
+    def __init__(self, context, pootle_path=None, fs_path=None):
+        self.context = context
```
Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: What is `context`?

Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: in this case its an FSPlugin, but its kinda agnostic, as long as the context has a resources attribute that (would if we had interfaces) implements ProjectResources (see class above)

Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: actually it needs `get_fs_path`and a couple of other things so it needs an `FSPlugin` really

Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: As long as there is some comment in code explaining what is needed I am happy.

**Comment by unho** [comment]
lgtm

File: pootle/apps/pootle_fs/resources.py [hunk 3]
```diff
@@ -47,3 +53,96 @@ def unsynced(self):
...
+    @cached_property
+    def fs_translations(self):
+        return sorted(self.context.find_translations(
+            fs_path=self.fs_path, pootle_path=self.pootle_path))
+
+    @cached_property
+    def resources(self):
+        return self.context.resources
+
+    @cached_property
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/resources.py]: why do we need `@cached_property` here?

File: pootle/apps/pootle_fs/resources.py [hunk 4]
```diff
@@ -47,3 +53,96 @@ def unsynced(self):
...
+        if not self.fs_path or fnmatch(path, self.fs_path):
+            return path
+
+    def _exclude_staged(self, qs):
+        return (
+            qs.exclude(staged_for_removal=True)
+              .exclude(staged_for_merge=True))
+
+    @cached_property
+    def fs_translations(self):
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/resources.py]: what are translations? localized files?

Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: hmm, yep - this defo needs to be renamed

its paths to found translation files

File: pootle/apps/pootle_fs/resources.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: this filter is called repeatedly - there is no reason to keep recreating it and the regexes etc

File: pootle/apps/pootle_fs/resources.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_fs/resources.py]: Addressed.

**Comment by unho** [comment]
lgtm

File: pootle/apps/pootle_fs/resources.py [hunk 5]
```diff
@@ -47,3 +53,111 @@ def unsynced(self):
...
+    If present all resources are filtered accordingly.
+    """
+
+    def __init__(self, context, pootle_path=None, fs_path=None):
+        self.context = context
+        self.pootle_path = pootle_path
+        self.fs_path = fs_path
+
+    def match_fs_path(self, path):
+        """Filter fs_paths using file glob"""
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/resources.py]: It seems "filter" is not a proper word here. Check probs?

Comment (phlax) [inline pootle/apps/pootle_fs/resources.py]: hmm well it does filter in the sense that it only returns if matched (or no match required)

**Comment by ta2-1** [comment]
lgtm

**Pull request merged**